src/fparser/two/tests/fortran2008/test_submodule_stmt_r1117.py

```diff
@@ -41,89 +41,98 @@
     assert "Submodule_Stmt: 'submod (id) name" in str(excinfo.value)
 
 
 def test_simple_error2():
     ''' Test the parsing of a submodule statement'''
+    _ = ParserFactory().create(std="f2008")
     with pytest.raises(NoMatchError) as excinfo:
-        _ = Submodule_Stmt("submodule name")
+        dummy_ = Submodule_Stmt("submodule name")
     assert "Submodule_Stmt: 'submodule name'" in str(excinfo.value)
 
 
 def test_simple_error3():
     ''' Test the parsing of a submodule statement'''
+    _ = ParserFactory().create(std="f2008")
     with pytest.raises(NoMatchError) as excinfo:
-        _ = Submodule_Stmt("submodule () name")
+        dummy_ = Submodule_Stmt("submodule () name")
     assert "Submodule_Stmt: 'submodule () name'" in str(excinfo.value)
 
 
 def test_simple_error4():
     ''' Test the parsing of a submodule statement'''
+    _ = ParserFactory().create(std="f2008")
     with pytest.raises(NoMatchError) as excinfo:
-        _ = Submodule_Stmt("submodule (id)")
+        dummy_ = Submodule_Stmt("submodule (id)")
     assert "Submodule_Stmt: 'submodule (id)'" in str(excinfo.value)
 
 
 def test_simple_error5():
     ''' Test the parsing of a submodule statement'''
+    _ = ParserFactory().create(std="f2008")
     with pytest.raises(NoMatchError) as excinfo:
-        _ = Submodule_Stmt("submodule name (id)")
+        dummy_ = Submodule_Stmt("submodule name (id)")
     assert "Submodule_Stmt: 'submodule name (id)'" in str(excinfo.value)
 
 
 def test_simple_error6():
     ''' Test the parsing of a submodule statement'''
+    _ = ParserFactory().create(std="f2008")
     with pytest.raises(NoMatchError) as excinfo:
-        _ = Submodule_Stmt("submodule (id) (name)")
+        dummy_ = Submodule_Stmt("submodule (id) (name)")
     assert "Submodule_Stmt: 'submodule (id) (name)'" in str(excinfo.value)
 
 
 def test_simple_error7():
     ''' Test the parsing of a submodule statement'''
+    _ = ParserFactory().create(std="f2008")
     with pytest.raises(NoMatchError) as excinfo:
-        _ = Submodule_Stmt("submodule id) name")
+        dummy_ = Submodule_Stmt("submodule id) name")
     assert "Submodule_Stmt: 'submodule id) name'" in str(excinfo.value)
 
+
 def test_simple_error8():
     ''' Test the parsing of a submodule statement'''
+    _ = ParserFactory().create(std="f2008")
     with pytest.raises(NoMatchError) as excinfo:
-        _ = Submodule_Stmt("submodule (id name")
+        dummy_ = Submodule_Stmt("submodule (id name")
     assert "Submodule_Stmt: 'submodule (id name'" in str(excinfo.value)
 
 
-@pytest.mark.xfail(reason="monkeypatch does not seem to work in this case")
 def test_splitparen_error(monkeypatch):
     '''Test that if the first argument of the splitparen is not empty then
     an error is returned.
 
     '''
+    _ = ParserFactory().create(std="f2008")
+
     def mockreturn(my_string):
-        return ["XXX", "", ""]
+        ''' Mock function for the splitparen function.'''
```

Comment:
Rather than define `mockreturn()`, you could just use a lambda for this:

    lambda x: ["XXX", "", ""]

but it's not a big deal.

Reply:
Not a big deal but the code is simpler so I've changed it. Thanks.